app/views/course/assessment/answer/programming/_programming.json.jbuilder

```diff
@@ -40,14 +40,15 @@ displayed_test_case_types << 'evaluation_test' if show_evaluation
 
 json.testCases do
   displayed_test_case_types.each do |test_case_type|
+    show_public_testcases_output = (test_case_type == "public_test") && current_course.show_public_test_cases_output
```

Comment:
Small issue, it's better to assign the whole expression to a single variable here. so when `can_grade` is `true`, the part after `||` is not evaluated.

Also for strings, we perfer `''` over `""`.
```
show_testcase_outputs = can_grade || (test_case_type == 'public_test' && current_course.show_public_test_cases_output)
```

Reply:
Sure will update the PR. I had put it in a completely separate variable, to increase the readability further. Will make the suggested changes.